profiling/src/collections/string_table/mod.rs

```diff
@@ -0,0 +1,279 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+#[allow(unused)]
+pub mod wordpress_test_data;
+
+use crate::collections::identifiable::{Id, StringId};
+use crate::iter::{IntoLendingIterator, LendingIterator};
+use datadog_alloc::{AllocError, Allocator, ChainAllocator, VirtualAllocator};
+use std::alloc::Layout;
+
+/// A trait that indicates an allocator is arena allocator, meaning it doesn't
+/// deallocate individual items, but deallocates their memory as a  group when
+/// the arena is dropped.
+pub trait ArenaAllocator: Allocator {
+    /// Copies the str into the arena, and returns a slice to the new str.
+    fn allocate(&self, str: &str) -> Result<&str, AllocError> {
```

Comment:
`allocate` is a funny name for this

Reply:
Have any ideas for a better one? If I had a better one, I'd have used it ^_^